front: fix various bugs related to PSLs edition

jacomyal opened the pull request:
This PR contains 3 commits that fix tickets:
  - [x] #4975 
  - [x] #4982 
  - [x] #5789 
  - [x] #5790

jacomyal committed:
front: hides geojson when editing a psl

Fix #5790.

jacomyal committed:
front: updates adaptFilter to write proper filters

Fix #5789.

In some case (editing PSL, at the moment), the filter given to
adaptFilter can be a logical tree containing some
ExpressionFilterSpecification, instead of LegacyFilterSpecification. In
those cases, the filters added by adaptFilter should be
ExpressionFilterSpecification as well, so that MapLibre does not log
tons of errors.

jacomyal committed:
front: fixes panels positions when editing psl

Fix #4982.
Fix #4975.

When editing panels positions of a speed section in the infra editor,
the positions where not properly interpolated between the computed
length of the track and its data length. Read #4547 for more
information.

flomonster reviewed (commented):
I've tested it and found out few bugs/irritants

- **Bug**: We cannot save a newly created PSL with panels. Editoast expects some kp but we cannot provide them through the form.
- **Bug**: When editing a speed limit some track sections can't be added or removed from the speed limit. I didn't find the reasons why these tracks couldn't be selected.
- **Irritant**: There is no way to see which speed section is currently edited.

Note: I don't know if these points need to be fixed by this PR. We can open new tickets if you wish.

SharglutDev commented on front/src/common/Map/Layers/GeoJSONs.tsx:
```diff
@@ -85,12 +120,23 @@ function adaptFilter(
+  } else {
+    // Add the conditions as LegacyFilterSpecification:
+    if (whiteList.length) conditions.push(['in', fieldId, ...whiteList]);
+    if (blackList.length) conditions.push(['!in', fieldId, ...blackList]);
```
It's not related to your PR but as we are on a open source project (and anyhow), I think we should use another syntax than "whitelist" and "blacklist"

jacomyal commented:
> **Bug**: We cannot save a newly created PSL with panels. Editoast expects some kp but we cannot provide them through the form.

New ticket #5792 will be handled outside this PR, since it involves back as well.

> **Bug**: When editing a speed limit some track sections can't be added or removed from the speed limit. I didn't find the reasons why these tracks couldn't be selected.

Ticket #5800 will be added to this PR if I find a fix before it's reviewed.

> **Irritant**: There is no way to see which speed section is currently edited.

Ticket #5799 will be added to this PR if I find a fix before it's reviewed.

SharglutDev reviewed (changes requested):
Thank you for the PR.

Left one comment and you have some warnings left in the code.

jacomyal committed:
front: cleans some infra edition related code

This fixes some comments on PR #5791.

Details:
- Renames whiteList and blackList from GeoJSONs.tsx, and adds comments
  to explicitely clarify what they are used for
- Cleans two linting warnings

SharglutDev reviewed (approved):
Thank you for the changes and comments on the function. Lgtm !

clarani reviewed (approved):
LGTM, tested :white_check_mark:
